### PR TITLE
Wifi upgrade

### PR DESCRIPTION
**Pull request by kuzeyron**
This have been my main focus.
Less Sudo and packages!

### CODE REVIEW

File: examples/wifi/main.py [hunk 1]
```diff
@@ -106,8 +106,12 @@ def show_wifi_scans(self):
         for name in wifi_scans:
             content = ""
             items = wifi._get_network_info(name)
-            for key, value in items.iteritems():
-                content += "{}:    {} \n".format(key, value)
+            if sys.version_info.major is 3: #Support for Py2 and Py3
+                for key, value in items.items():
+                    content += "{}:    {} \n".format(key, value)
+            else:
+                for key, value in items.iteritems():
```
Comment (inclement) [inline examples/wifi/main.py]: I would just use `items.items()` to keep things simple, it seems unlikely that the dict will have enough keys for it to matter in Python 2.

File: plyer/platforms/linux/wifi.py [hunk 2]
```diff
@@ -4,37 +4,76 @@
         - nmcli (Network Manager command line tool)
             It is found in most of the popular distros. Support for other
             managers is not provided yet.
-
-        - python-wifi module
-            `https://wifi.readthedocs.io/en/latest/`
-            `https://github.com/rockymeza/wifi`
-
+    One more thing: Betalistish!!
```
Comment (inclement) [inline plyer/platforms/linux/wifi.py]: Did you mean to include this?

File: plyer/platforms/linux/wifi.py [hunk 3]
```diff
@@ -73,26 +112,24 @@ def _connect(self, network, parameters):
...
     import sys
     try:
-        import wifi
         return LinuxWifi()
     except ImportError:
-        sys.stderr.write("python-wifi not installed. try:"
-                         "`pip install --user wifi`.")
-
-    return Wifi()
+        sys.stderr.write("Oh look a penny..")
```
Comment (inclement) [inline plyer/platforms/linux/wifi.py]: Did you mean to include this?

File: examples/wifi/main.py (hunk 1 above)
Comment (kuzeyron) [reply examples/wifi/main.py]: I think I tried using only items.items(). But if someone have time to clean my mess I'd be more than happy!

File: plyer/platforms/linux/wifi.py (hunk 2 above)
Comment (kuzeyron) [reply plyer/platforms/linux/wifi.py]: Only for beta testers! There are some jobs left to do. But beta testers are needed right now!

**Comment by kuzeyron** [comment]
Changed the code to work better.
Source taken from: https://cgit.freedesktop.org/NetworkManager/NetworkManager/tree/examples/python/gi/show-wifi-networks.py

File: plyer/platforms/linux/wifi.py [hunk 4]
```diff
@@ -1,98 +1,186 @@
...
+		- nmcli (Network Manager command line tool)
+			It is found in most of the popular distros. Support for other
+			managers is not provided yet.
+			Source from: https://cgit.freedesktop.org/NetworkManager/NetworkManager/tree/examples/python/gi/show-wifi-networks.py
 '''
-
+#from __future__ import print_function
 from plyer.facades import Wifi
 from subprocess import Popen, PIPE, call, STDOUT
+import gi
```
Comment (akshayaurora) [inline plyer/platforms/linux/wifi.py]: This is a strict no, we can not just import a gui lib for managing wifi `nmcli` is not dependent on the gui lib. Think of distress that have no gnome libs at all.

**Comment by akshayaurora** [comment]
closing due to dependency on go

**Pull request closed without merging**